text/0000-prepare-operation.md

```diff
@@ -0,0 +1,387 @@
+
+# Definitions
+[definitions]: #definitions
+
+- namespace - A table within `project.toml` that encompasses a set of properties.
+- Prepare operation - a new operation that occurs before the Build operation.
+- `project.toml` - A project descriptor file specified [here][project-descriptor-spec].
+
+[project-descriptor-spec]: https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md
+
+# Motivation
+[motivation]: #motivation
+
+The motivation for this change was to enable the following goals from being achieved:
+
+  - **Goal 1: Serializing CLI configuration**
+    > As a user, I would like to be able to serialize and share the parameters I use with certain platforms such as `pack`.
+  - **Goal 2: A recognizable file in repositories**
+    > As a user, I would like to be able to recognize, based on the file system, if a project is using Cloud Native Buildpacks.
+  - **Goal 3: Platform recognition of project.toml**
+    > As a user, I would like to ensure that my configuration in `project.toml` is being used.
+
+**This RFC directly solves for Goal 3**, while enabling the solutions for Goals 1 and 2 via a supplemental RFC ([Support for pack.toml][rfc-pr-189]).
+
+[rfc-pr-189]: https://github.com/buildpacks/rfcs/pull/189
+
+# What it is
+[what-it-is]: #what-it-is
+
+The proposal is composed of following changes:
+
+- [Changes to `io.buildpacks` namespace.](#changes-to-iobuildpacks-namespace)
+  - [Define criteria for properties in namespace.](#criteria-for-properties)
+  - [Version `io.buildpacks` namespace.](#version-iobuildpacks-namespace)
+- [A replaceable new phase `prepare` that applies platform configuration.](#prepare-phase)
+- [Supporting Cloud Native Buildpack utilities.](#cloud-native-buildpacks-utilities)
+
+## Changes to `io.buildpacks` namespace
+
+### Criteria for properties
+
+This namespace declares default values that platforms should acknowledge. A majority of properties should be spec'd but we would allow for unspec'd additional properties.
+
+The criteria for spec'd properties in this namespace would be based on whether they affect inputs to the lifecycle phases. For example, the order and environment variables are inputs to the lifecycle. The `builder`, in contrast, is not an input to the lifecycle so it wouldn't meet the criteria.
+
+Therefore the following changes have been made from the [latest schema][pd-02]:
+
+ - **Remove `builder` from `io.buildpacks` namespace**
```

Comment:
Does `pack` support this key today? I think my vote would be for a platform to fail building (or warn, I guess) if the builder key isn't supported. But if we remove it entirely I suppose I won't be too upset. Platforms can use their own namespace to achieve the same thing.